nvflare/private/fed/server/job_runner.py

```diff
@@ -344,67 +345,66 @@ def run(self, fl_ctx: FLContext):
                     )
 
                     if ready_job:
-                        with self.lock:
-                            client_sites = {k: v for k, v in sites.items() if k != "server"}
-                            job_id = None
-                            try:
-                                self.log_info(fl_ctx, f"Got the job:{ready_job.job_id} from the scheduler to run")
-                                fl_ctx.set_prop(FLContextKey.CURRENT_JOB_ID, ready_job.job_id)
-                                job_id, failed_clients = self._deploy_job(ready_job, sites, fl_ctx)
-                                job_manager.set_status(ready_job.job_id, RunStatus.DISPATCHED, fl_ctx)
-
-                                deploy_detail = fl_ctx.get_prop(FLContextKey.JOB_DEPLOY_DETAIL)
-                                if deploy_detail:
-                                    job_manager.update_meta(
-                                        ready_job.job_id,
-                                        {
-                                            JobMetaKey.JOB_DEPLOY_DETAIL.value: deploy_detail,
-                                            JobMetaKey.SCHEDULE_COUNT.value: ready_job.meta[
-                                                JobMetaKey.SCHEDULE_COUNT.value
-                                            ],
-                                            JobMetaKey.LAST_SCHEDULE_TIME.value: ready_job.meta[
-                                                JobMetaKey.LAST_SCHEDULE_TIME.value
-                                            ],
-                                            JobMetaKey.SCHEDULE_HISTORY.value: ready_job.meta[
-                                                JobMetaKey.SCHEDULE_HISTORY.value
-                                            ],
-                                        },
-                                        fl_ctx,
-                                    )
-
-                                if failed_clients:
-                                    deployable_clients = {
-                                        k: v for k, v in client_sites.items() if k not in failed_clients
-                                    }
-                                else:
-                                    deployable_clients = client_sites
-
-                                self._start_run(
-                                    job_id=job_id,
-                                    job=ready_job,
-                                    client_sites=deployable_clients,
-                                    fl_ctx=fl_ctx,
+                        # with self.lock:
```

Comment:
If we remove this lock, do we need to make sure the job_manager will function correctly if multiple threads calling it?

Reply:
The self.lock is not totally removed. Just reduced the lock block scope. Made it less impact other codes executing.